sdk/javascript/src/symbol/models_ts.js

```diff
@@ -0,0 +1,1039 @@
+ */
+export class AddressAliasTransactionV1Descriptor {
+	/**
+	 * Creates a descriptor for AddressAliasTransactionV1.
+	 * @param {models.NamespaceId} namespaceId Identifier of the namespace that will become (or stop being) an alias for the address.
+	 * @param {Address} address Aliased address.
+	 * @param {models.AliasAction} aliasAction Alias action.
+	 */
+	constructor(namespaceId, address, aliasAction) {
+		this.rawDescriptor = {
+			type: 'address_alias_transaction_v1',
+			namespaceId,
+			address,
+			aliasAction
+		};
+	}
+
+	/**
+	 * Builds a representation of this descriptor that can be passed to a factory function.
+	 * @returns {object} Descriptor that can be passed to a factory function.
+	 */
+	toMap() {
+		return this.rawDescriptor;
+	}
+}
+/**
+ * Type safe descriptor used to generate a descriptor map for MosaicAliasTransactionV1Descriptor.
+ *
+ * Attach or detach a [namespace](/concepts/namespace.html) to a Mosaic.(V1, latest)
+ * Setting an alias to a mosaic is only possible if the account announcing this transaction has also created the namespace and the mosaic involved.
+ */
+export class MosaicAliasTransactionV1Descriptor {
+	/**
+	 * Creates a descriptor for MosaicAliasTransactionV1.
+	 * @param {models.NamespaceId} namespaceId Identifier of the namespace that will become (or stop being) an alias for the Mosaic.
+	 * @param {models.MosaicId} mosaicId Aliased mosaic identifier.
+	 * @param {models.AliasAction} aliasAction Alias action.
+	 */
+	constructor(namespaceId, mosaicId, aliasAction) {
+		this.rawDescriptor = {
+			type: 'mosaic_alias_transaction_v1',
+			namespaceId,
+			mosaicId,
+			aliasAction
+		};
+	}
+
+	/**
+	 * Builds a representation of this descriptor that can be passed to a factory function.
+	 * @returns {object} Descriptor that can be passed to a factory function.
+	 */
+	toMap() {
+		return this.rawDescriptor;
+	}
+}
+/**
+ * Type safe descriptor used to generate a descriptor map for NamespaceRegistrationTransactionV1Descriptor.
+ *
+ * Register (or renew a registration for) a [namespace](/concepts/namespace.html) (V1, latest).
+ * Namespaces help keep assets organized.
+ */
+export class NamespaceRegistrationTransactionV1Descriptor {
+	/**
+	 * Creates a descriptor for NamespaceRegistrationTransactionV1.
+	 * @param {models.NamespaceId} id Namespace identifier.
+	 * @param {models.NamespaceRegistrationType} registrationType Namespace registration type.
+	 * @param {models.BlockDuration|undefined} duration Number of confirmed blocks you would like to rent the namespace for. Required for root namespaces.
+	 * @param {models.NamespaceId|undefined} parentId Parent namespace identifier. Required for sub-namespaces.
+	 * @param {Uint8Array|string|undefined} name Namespace name.
+	 */
+	constructor(id, registrationType, duration = undefined, parentId = undefined, name = undefined) {
```

Comment:
Should there be a constructor for creating the root vs sub namespace? This would ensure that the correct values are passed in.

Reply:
good suggestion, i will look into this

Reply:
since JS doesn't support overloaded ctors, we could potentially add factory functions.

however, that probably doesn't make sense because it would be a different paradigm just for that one class, and the main goal of this is to simplify and streamline usage.